src/rpc/connection.rs

```diff
@@ -685,13 +667,48 @@ impl CircularQueue {
     }
 }
 
+pub fn clean_up(queue: Arc<CircularQueue>) {
+    debug!("start to cleanup");
+    loop {
+        let mut idx = queue.start_index.load(Ordering::Acquire);
+        let end_flag = queue.end_index.load(Ordering::Acquire);
```

Comment:
Mod `idx` and `end_flag` by REQUEST_QUEUE_LENGTH here may help.

Reply:
there is no need to mod.
the sub `end_flag - idx` contains information of how many callbacks are working.
If the sub is multiple times of REQUEST_QUEUE_LENGTH, we could know how many callbacks are blocking.